File: Java/src/main/java/com/foodcall/domain/entities/Category.java
package com.foodcall.domain.entities;

import com.foodcall.domain.exceptions.InvalidEntityException;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Entity
@Table(name = "categories")
@Getter
@NoArgsConstructor
public class Category {
    
    @Id
    private UUID id;
    
    @Column(nullable = false, length = 50)
    private String name;
    
    @Column(length = 200)
    private String description;
    
    @Column(nullable = false)
    private boolean isActive;
    
    public Category(String name, String description) {
        validateName(name);
        validateDescription(description);
        
        this.id = UUID.randomUUID();
        this.name = name;
        this.description = description;
        this.isActive = true;
    }
    
    public void updateName(String name) {
        validateName(name);
        this.name = name;
    }
    
    public void updateDescription(String description) {
        validateDescription(description);
        this.description = description;
    }
    
    public void activate() {
        this.isActive = true;
    }
    
    public void deactivate() {
        this.isActive = false;
    }
    
    private void validateName(String name) {
        if (name == null || name.trim().isEmpty()) {
            throw new InvalidEntityException("Category", "Nome não pode ser vazio");
        }
        if (name.length() < 2) {
            throw new InvalidEntityException("Category", "Nome deve ter pelo menos 2 caracteres");
        }
        if (name.length() > 50) {
            throw new InvalidEntityException("Category", "Nome não pode ter mais de 50 caracteres");
        }
    }
    
    private void validateDescription(String description) {
        if (description != null && !description.trim().isEmpty() && description.length() > 200) {
            throw new InvalidEntityException("Category", "Descrição não pode ter mais de 200 caracteres");
        }
    }
}

File: NET_8/FoodCall.Domain/Entities/Category.cs
using FoodCall.Domain.Exceptions;


namespace FoodCall.Domain.Entities
{
    public class Category
    {
        public Guid Id { get; private set; }
        public string Name { get; private set; }

        public string Description { get; private set; }
        public bool IsActive { get; private set; }
   

    public Category(string name, string description)
    {
        ValidateName(name);
        ValidateDescription(description);

        Id = Guid.NewGuid();
        Name = name;
        Description = description;
        IsActive = true;
    }

    public void UpdateName(string name)
    {
        ValidateName(name);
        Name = name;
    }

    public void UpdateDescription(string description)
    {
        ValidateDescription(description);
        Description = description;
    }

    public void Activate()
    {
        IsActive = true;
    }

    public void Deactivate()
    {
        IsActive = false;
    }

    // Validações privadas - regras de negócio da entidade
        private void ValidateName(string name)
        {
            if (string.IsNullOrWhiteSpace(name))
                throw new InvalidEntityException("Category", "Nome não pode ser vazio");

            if (name.Length < 2)
                throw new InvalidEntityException("Category", "Nome deve ter pelo menos 2 caracteres");

            if (name.Length > 50)
                throw new InvalidEntityException("Category", "Nome não pode ter mais de 50 caracteres");
        }

        private void ValidateDescription(string description)
        {
            // Descrição é opcional
            if (string.IsNullOrWhiteSpace(description))
                return;

            if (description.Length > 200)
                throw new InvalidEntityException("Category", "Descrição não pode ter mais de 200 caracteres");
        }
    } 
 }   